### ISSUE TITLE
Grid mode auto-zoom should be improved

### ISSUE THREAD

**Issue by jni** [napari/napari#1689]
## 🚀 Feature

When using grid view with just two layers, the zoom is set as if I was looking at a 4x4 grid. imho it should be a bit smarter and zoom in further. Here's a small example to reproduce:

```
import numpy as np
import napari


n = 28
image = np.random.random((2, n, n))

with napari.gui_qt():
    viewer = napari.view_image(image, channel_axis=0)
    viewer.grid_view()
```

This looks like this:

<img width="1200" alt="Screen Shot 2020-10-04 at 1 21 59 pm" src="https://user-images.githubusercontent.com/492549/95005626-ef779100-0646-11eb-9904-1e9fdd6009c9.png">

Whereas I think it should look like this:

<img width="1200" alt="Screen Shot 2020-10-04 at 1 22 16 pm" src="https://user-images.githubusercontent.com/492549/95005638-f9998f80-0646-11eb-8f44-9f3865cffb87.png">

**Comment by sofroniewn** [comment]
I can fix this when I get back to the camera PR #854, thanks for the flag!! All this stuff is becoming much much easier to reason about with the merge of #1660 and things like #1673 getting close!!

**Comment by sofroniewn** [comment]
Actually I might wait till after #854 to fix it in its own PR where I clean up grid mode, that's actually been long overdue. Regardless this should get fixed soon, and the screenshots are very helpful!!

**Comment by GenevieveBuckley** [comment]
Speaking of auto-zoom, should we have this happen every time a new layer is added? And does auto-zoom take into account all of the layers, or all of the visible layers, or just the last selected layer?

If I add an image with a larger field of view after one with a smaller field of view, then parts of the bigger image appear cut off in the image viewer. It's not always super-obvious the user just needs to zoom in or out, if the cut off part isn't a lot I can wind up wondering what's gone wrong with my data.

**Comment by sofroniewn** [comment]
Ok I'm looking at this again. Funnily enough the discussion happening at the tail end of https://github.com/napari/napari/pull/854#issuecomment-713843658 is highly relevant here too due to the non-square aspect ratio of the grid example above. I can try and improve things, but wanted to link there too.


> Speaking of auto-zoom, should we have this happen every time a new layer is added? And does auto-zoom take into account all of the layers, or all of the visible layers, or just the last selected layer?

- Right reset view does not happen automatically when a layer adds
- Reset view takes into account all layers, including non-visible and non-selected

> If I add an image with a larger field of view after one with a smaller field of view, then parts of the bigger image appear cut off in the image viewer. It's not always super-obvious the user just needs to zoom in or out, if the cut off part isn't a lot I can wind up wondering what's gone wrong with my data.

Hmm ok, yeah we might want to reset the view when people add new layers, might be more natural.
